backend/src/app.service.ts

```diff
@@ -0,0 +1,88 @@
+import { Injectable } from '@nestjs/common';
+import { Neo4jService } from 'nest-neo4j/dist';
+import { Node } from './entities/Node';
+import { EdgeDescriptor } from './entities/EdgeDescriptor';
+import { Edge } from './entities/Edge';
+import { QueryResult } from './entities/queries/QueryResult';
+import { LimitQuery } from './entities/queries/LimitQuery';
+import { NodeDescriptor } from './entities/NodeDescriptor';
+
+@Injectable()
+export class AppService {
+  constructor(private readonly neo4jService: Neo4jService) {}
+
+  async queryAll(query?: LimitQuery): Promise<QueryResult> {
+    return {
+      edges: query?.limit?.edges === 0 ? [] : await this.getAllEdges(query),
+      nodes: query?.limit?.nodes === 0 ? [] : await this.getAllNodes(query),
+    };
+  }
+
+  /**
+   * Returns a list the ids of all nodes
+   */
+  private async getAllNodes(query?: LimitQuery): Promise<NodeDescriptor[]> {
+    // toInteger required, since apparently it converts int to double...
+    const result = await this.neo4jService.read(
+      `
+      MATCH (n)
+      RETURN ID(n) as id
+      ${query?.limit?.nodes ? 'LIMIT toInteger($limitNodes)' : ''}
+    `,
+      {
+        limitNodes: query?.limit?.nodes,
+      },
+    );
+    return result.records.map((x) => x.toObject() as NodeDescriptor);
+  }
+
+  /**
+   * Returns list of Nodes
+   */
+  async getNodesById(ids: number[]): Promise<Node[]> {
+    const result = await this.neo4jService.read(
+      'MATCH (n) WHERE ID(n) IN $ids RETURN ID(n) as id, labels(n) as labels, properties(n) as properties',
+      { ids },
+    );
+
+    return result.records.map((record) => record.toObject() as Node);
+  }
+
+  /**
+   * Returns a list the ids of all edges
+   */
+  private async getAllEdges(query?: LimitQuery): Promise<EdgeDescriptor[]> {
```

Comment:
Why does this receive the complete query, if it only needs to know the limit of edge count?